modules/aac/sdp.c

```diff
@@ -26,30 +28,64 @@ static unsigned param_value(const char *fmtp, const char *name)
 }
 
 
-int aac_fmtp_enc(struct mbuf *mb, const struct sdp_format *fmt,
-		 bool offer, void *arg)
+/* check decoding compatibility of remote format */
+bool aac_fmtp_cmp(const char *lfmtp, const char *rfmtp, void *arg)
 {
-	(void)offer;
+	(void)lfmtp;
 	(void)arg;
 
-	if (!mb || !fmt)
-		return 0;
+	uint32_t plid;
 
-	return mbuf_printf(mb, "a=fmtp:%s "
-			   "profile-level-id=24;object=%i;bitrate=%u\r\n",
-			   fmt->id, AOT_ER_AAC_LD, AAC_BITRATE);
-}
+	struct pl pl, val;
 
+	if (!rfmtp)
+		return false;
 
-bool aac_fmtp_cmp(const char *lfmtp, const char *rfmtp, void *arg)
-{
-	(void)lfmtp;
-	(void)arg;
+	pl_set_str(&pl, rfmtp);
+
+	debug("aac: compare: %s\n", rfmtp);
+
+	if (fmt_param_get(&pl, "mode", &val)) {
+		if (strncasecmp("AAC-hbr", val.p, val.l))
+			return false;
+	}
+
+	if (param_value(rfmtp, "streamType") != 5)
+		return false;
+
+	if (param_value(rfmtp, "sizeLength") != SIZELENGTH)
+		return false;
+
+	if (param_value(rfmtp, "indexLength") != INDEXLENGTH)
+		return false;
+
+	if (param_value(rfmtp, "indexDeltaLength") != INDEXDELTALENGTH)
+		return false;
+
+	if (param_value(rfmtp, "bitrate") < 8000 ||
+	    param_value(rfmtp, "bitrate") > 576000)
+		return false;
```

Comment:
is the `bitrate` parameter defined in RFC 3640 ?

if this parameter is missing, a sane default value should be used.

Reply:
Mmmmh...
I did not think about it at anytime as all of my test devices used it.

But if `bitrate` wont be there things may even get more complicated.

Reply:
Should be no problem to make the bitrate fmtp-property optional.
I think I remember that FDK-AAC will automatically select a proper bitrate if it is not specified explicit.